extensions/wikia/WikiFactory/WikiFactory.php

```diff
@@ -1192,11 +1212,12 @@ public static function getCurrentStagingHost( $dbName='', $default='', $host = n
 	 * @author pbablok@wikia
 	 * @static
 	 *
-	 * @param string $url	general URL pointing to any server
-	 *
-	 * @return string	url pointing to local env
+	 * @param string $url general URL pointing to any server
+	 * @param null $forcedEnv go get url for env we want to
+	 * @return string url pointing to local env
+	 * @throws \Exception
 	 */
-	static public function getLocalEnvURL( $url ) {
+	static public function getLocalEnvURL( $url, $forcedEnv = null ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this function "getLocalEnvURL" is 22 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 7, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)